src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/ClusterAwareWriterFailoverHandler.java

```diff
@@ -298,7 +298,7 @@ public WaitForNewWriterHandler(
       this.originalWriterHost = currentHost;
     }
 
-    public WriterFailoverResult call() {
+    public WriterFailoverResult call() throws SQLException {
```

Comment:
I'd prefer to catch and handle exceptions inside this task. Bubbling up exception may have side effects on entire failover process.